server/utils/cloudFunctions/functions/test/algoliaTests/algoliaFunctions.test.js

```diff
@@ -0,0 +1,143 @@
+/**
+ * Description: Unit tests for algoliaFunctions.js
+ *              Uses stubbing to mock algolia save/delete.
+ *              Uses sinon/chai to spy on calls and validate behavior.
+ *              compile with `npm run test`.
+ * Author: Robert Marcus
+ * Date: July 11, 2020
+ */
+
+// Import the testing environment configuration
+const { test, assert, sinon } = require('../testConfig');
+
+// Import the generic cloud functions.
+const devAlgoliaFunctions = 
+    require('../../algoliaCloudFunctions/devAlgoliaFunctions');
+const prodAlgoliaFunctions = 
```

Comment:
Why do you have prod stuff in here?  Unless there's a good reason for it, it's incredibly dangerous to be messing with prod in your tests.  What if you wrote the test wrong and end up actually writing to prod?  What if some future engineer comes along and tinkers with the test and accidentally writes to prod?

Is there something you need to test with prod that you can't test with devel?

Reply:
Good points, had not considered the danger of prod. First instinct was just to have a test for it, but I suppose since it is supposed to just be a mirror of prod, the onus is on the author to ensure synchronicity, not the tests.